apps/analysis/serializers.py

```diff
@@ -130,6 +138,18 @@ def get_entries_used_in_analysis(self, analysis):
             analysis_pillar__analysis=analysis
         ).annotate(entries_in_analysis=models.Count('entries')).filter(entries_in_analysis__gt=0).count()
 
+    def get_total_sources(self, analysis):
+        return Lead.objects.filter(
+            project=analysis.project
+        ).annotate(
+            entries_count=models.Count('entry')
+        ).filter(entries_count__gt=0).count()
+
+    def get_total_entries(self, analysis):
+        return Entry.objects.filter(
+            project=analysis.project
+        ).count()
+
```

Comment:
Looks like we are doing this calculation for a queryest. Change this to avoid n+1 issue.

Reply:
Nice point. Lets annotate all of these fields during the queryset calculation.

Plus, should we ignore the discarded entries here?